processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -1474,7 +1474,15 @@ public Rowboat apply(Rowboat lhs, Rowboat rhs)
       Object[] rhsMetrics = rhs.getMetrics();
 
       for (int i = 0; i < metrics.length; ++i) {
-        metrics[i] = metricAggs[i].combine(lhsMetrics[i], rhsMetrics[i]);
+        Object lhsMetric = lhsMetrics[i];
+        Object rhsMetric = rhsMetrics[i];
+        if (lhsMetric == null) {
+          metrics[i] = rhsMetric;
+        } else if (rhsMetric == null) {
+          metrics[i] = lhsMetric;
```

Comment:
@xvrl yes this case does occur. I have added a unit test which triggers this condition.  This was the main point of discussion in the issue, and the part that I needed most feedback on.

In the case were both are null, the value of metrics[i] = null.  This does not seem to a be a problem during the merge, and most Metric Serializers check for null before writing out a value. However, I do see how this could be a little funky, and does then require that serializers are not going to blow up, or produce incorrect results.